Classes/Service/SessionHandler.php

```diff
@@ -12,17 +12,24 @@
 use Extcode\Cart\Domain\Model\Cart\Cart;
 use Extcode\Cart\Domain\Model\Order\AbstractAddress;
 use TYPO3\CMS\Core\SingletonInterface;
+use TYPO3\CMS\Frontend\Authentication\FrontendUserAuthentication;
 
 class SessionHandler implements SingletonInterface
 {
     protected $prefixKey = 'cart_';
+    private FrontendUserAuthentication $feUser;
```

Comment:
The variable could also be renamed:
```suggestion
    private FrontendUserAuthentication $frontendUserAuthentication;
```

Reply:
Yeah, I thought as well that is does not fit but I did not want to change the original wording `$GLOBALS['TSFE']->fe_user`. But I think as well that this is much better.

Done.